### PR TITLE
feat!: allow body reuse

### PR DESCRIPTION
**Pull request by bobbyg603**
* Allow json body reuse
* Safer type for BugSplatResponse

BREAKING CHANGE: ApiClients now return Response so they can be cloned, BugSplatResponse json property is no longer type any.

### CODE REVIEW

File: spec/files/native/post-native-crash.ts [hunk 1]
```diff
@@ -39,7 +39,7 @@ export async function postNativeCrash(
         crashFile,
         'ebe24c1cd1a0912904658fa4fad2b539'
     );
-    const json = await postCrashResult.json();
+    const json = await postCrashResult.json() as { crashId: number };
```
Comment (bobbyg603) [inline spec/files/native/post-native-crash.ts:42]: not needed, remove

File: src/common/data/table-data/table-data-client/table-data-client.ts [hunk 2]
```diff
@@ -38,30 +38,34 @@ export class TableDataClient {
...
     return {
       status,
       json
     };
   }
 }
 
-
+// https://www.typescriptlang.org/docs/handbook/2/mapped-types.html#key-remapping-via-as
+// export type RawResponse<T, U = unknown> = Array<{ Rows: Array<T>, PageData: U }>;
```
Comment (bobbyg603) [inline src/common/data/table-data/table-data-client/table-data-client.ts:67]:
```suggestion
```

File: src/post/crash-post-client.e2e.ts [hunk 3]
```diff
@@ -20,7 +20,7 @@ describe('CrashPostClient', () => {
                 crashFile,
                 md5
             );
-            const json = await result.json();
+            const json = await result.json() as { crashId: number };
```
Comment (bobbyg603) [inline src/post/crash-post-client.e2e.ts:23]: no longer needed, remove

File: src/common/data/table-data/table-data-client/table-data-response.ts [hunk 4]
```diff
@@ -1,4 +1,4 @@
-export interface TableDataResponse<T> {
+export interface TableDataResponse<T, U = unknown> {
   rows: Array<T>;
-  pageData?: Record<string, unknown>;
+  pageData?: Record<string, U>;
```
Comment (bobbyg603) [inline src/common/data/table-data/table-data-client/table-data-response.ts:3]: this isn't quite right, pageData should be type U with default of Record<string, unknown>

**Review by daveplunkett — APPROVED** [review]

**Pull request merged**